iati/core/tests/test_rulesets.py

```diff
@@ -408,6 +403,64 @@ def test_rule_paths_more(self, basic_rule):
         else:
             assert basic_rule.more.startswith(basic_rule.xpath_base)
 
+    def test_incorrect_date_format_raises_error(self, rule):
+        """Check that a dataset with dates in an incorrect format raise expected error."""
+        with pytest.raises(ValueError):
+            rule.is_valid_for(iati.core.tests.utilities.DATASET_FOR_DATEORDER_RULE_INVALID_DATE_FORMAT)
+
+
+class TestRuleDependent(RuleSubclassTestBase):
+    """A container for tests relating to RuleDependent."""
+
+    @pytest.fixture
+    def rule_type(self):
+        """Type of rule."""
+        return 'dependent'
+
+    @pytest.fixture(params=[
+        {'paths': ['']},  # empty path
+        {'paths': ['path_1']},  # single path
+        {'paths': ['path_1', 'path_2']},  # multiple paths
+        {'paths': ['path_1', 'path_1']}  # duplicate paths
+    ])
+    def valid_case(self, request):
+        """Permitted case for this rule."""
+        return request.param
+
+    @pytest.fixture(params=[
+        {'paths': []},  # empty path array
+        {'paths': 'path_1'},  # non-array `paths`
+        {'paths': [3]},  # non-string value in path array
+        {'paths': ['path_1', 3]},  # mixed string and non-string value in path array
+        {},  # empty dictionary
+        {'paths': {'path_1'}}  # path value is dictionary instead of list
+    ])
+    def invalid_case(self, request):
+        """Non-permitted case for this rule."""
+        return request.param
+
+    @pytest.fixture
+    def invalid_dataset(self):
+        """Invalid dataset for this Rule."""
+        return iati.core.tests.utilities.DATASET_FOR_DEPENDENT_RULE_INVALID
+
+    @pytest.fixture
+    def valid_dataset(self):
+        """Return valid dataset for this Rule."""
+        return iati.core.tests.utilities.DATASET_FOR_DEPENDENT_RULE_VALID
+
+    @pytest.fixture(params=[
+        {"paths": ["transaction/provider-org", "location/point"]}
```

Comment:
valid_case is only for testing whether the Rule instantiates. It's not possible for all of them to validate a dataset, for example {'paths': ['']} is a valid path for instantiation but won't always return True for valid data because it won't always create a valid XPath search (this problem of empty paths applies to all Rules and is tested elsewhere).